subprojects/delivery/signer/src/main/kotlin/com/avito/plugin/SignServicePlugin.kt

```diff
@@ -42,65 +44,83 @@ public class SignServicePlugin : Plugin<Project> {
 
                 val urlResolver = UrlResolver(extension)
 
+                if (extension.apkSignTokens.containsKey(variant.name)) {
+                    target.tasks.register<SignApkTask>(signApkTaskName(variant.name)) {
+                        description = "Signs ${variant.name} apk with in-house service"
+                        apkDirectory.set(variant.artifacts.get(SingleArtifact.APK))
+                        tokenProperty.set(extension.apkSignTokens[variant.name]!!)
```

Comment:
We will have bad error message there

Reply:
Sorry I don't get that it's impossible

Reply:
i with i can express this better with kotlin.
Maybe some unwrap private function will increase readability